### PR TITLE
Add dynamic spec to servingruntime objects

### LINKED ISSUE #814: [Model Serving]: Dynamic spec of ServingRuntime
### Feature description

ServingRuntime creations is based in a [default template](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/api/network/servingRuntimes.ts#L23) with a harcoded image, we want to support:

1. Dynamic creation based on CRDs
2. Dynamic image fetching for the ServingRuntime container

## Dynamic creation based on CRDs
We wanna figure out a way to just include some options for modifying the actual spec of the object

## Dynamic image fetching for the ServingRuntime container
We might wanna retrieve this image dynamically from the operator.

Both of them need further investigation for the implementation.

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Fixes #814 

## How Has This Been Tested?

1. Deploy all the files in the manifest folder
2. Create a new Model Server
3. Check that the ServingRuntime is created with the specification of the `servingruntimes-config` ConfigMap.
4. Now use #806 to fake a regular user.
5. Repeate the same process
6. Ensure regular user with no read options can access the manifest too

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by DaoDaoNoCode — APPROVED** [review]
Code lgtm, didn't test on my cluster. We can wait until @andrewballantyne have another review on it.

File: frontend/src/api/network/servingRuntimes.ts
Comment (andrewballantyne) [inline frontend/src/api/network/servingRuntimes.ts:25]: If somehow you failed to have a default-config, would we not want to fallback on `DEFAULT_MODEL_SERVING_TEAMPLATE`?

```
YAML.parse('');
// null
```

File: frontend/src/api/network/servingRuntimes.ts
Comment (andrewballantyne) [inline frontend/src/api/network/servingRuntimes.ts:58]: There is no safety for `spec` not being there... In my previous comment, you'll spread `null` and then get a NPE on `.spec` when doing `.replicas`

Curious if you think just setting `servingRuntime` to `DEFAULT_MODEL_SERVING_TEAMPLATE` if it's null or undefined will just solve this?

File: manifests/modelserving/servingruntimes-config.yaml
Comment (andrewballantyne) [inline manifests/modelserving/servingruntimes-config.yaml:6]: Is there a reason we are providing this YAML? Isn't the whole point to fall back on ours and then if they WANT to, they can override it? Is this some sorta self documenting YAML?

I ask this because you're duplicating efforts here -- any change to the default needs to be done to this -- and they are not tied together.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/src/api/network/servingRuntimes.ts
Comment (lucferbux) [reply frontend/src/api/network/servingRuntimes.ts:25]: Yes, it was already getting to the `DEFAULT_MODEL_SERVING_TEAMPLATE ` but for the wrong reasons, I'm throwing an error to just trigger the fallback.

File: frontend/src/api/network/servingRuntimes.ts
Comment (lucferbux) [reply frontend/src/api/network/servingRuntimes.ts:36]: I'm addressing this more or less, but it's a fair nitpick, I'll change this

File: manifests/modelserving/servingruntimes-config.yaml
Comment (lucferbux) [reply manifests/modelserving/servingruntimes-config.yaml:6]: This yaml is only for upstream and it's not even included in the manifests, in downstream will be handle by the deployer.

File: frontend/src/api/network/servingRuntimes.ts
Comment (lucferbux) [reply frontend/src/api/network/servingRuntimes.ts:58]: mmmm this is a great discussion, I've addressed the issue with servingRuntime being null, but as now it will be a user configurable spec inside a configmap the user can misconfigured the object in many ways. With the current block, if there's a misconfiguration in the key spec elements (the ones we are checking in this function) it will fail and fallback to `DEFAULT_MODEL_SERVING_TEAMPLATE `.
But this is a tricky question, having a ConfigMap is not the best solution IMO, everything could go south since there's not "training wheels" for this feature, we might wanna check this in the future.

File: manifests/modelserving/servingruntimes-config.yaml
Comment (andrewballantyne) [reply manifests/modelserving/servingruntimes-config.yaml:6]: I stand by thinking this is a DRY issue -- but if you think it's needed, we can keep it

File: frontend/src/api/network/servingRuntimes.ts
Comment (andrewballantyne) [reply frontend/src/api/network/servingRuntimes.ts:64]: We'll need to log a bug to handle this more and surface it to the user more -- please log one -- I want to make it a blocker for 1.22 / aka our next ODH Dashboard release

Comment (lucferbux) [reply frontend/src/api/network/servingRuntimes.ts:64]: Done here: https://github.com/opendatahub-io/odh-dashboard/issues/893

File: frontend/src/api/network/servingRuntimes.ts
Comment (andrewballantyne) [inline frontend/src/api/network/servingRuntimes.ts:34]: Only concern I have after testing this back and forth is -- this is going to be hard to see as an error in prod -- any concerns about making it a `console.error`? I don't think it should cause any issues.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**
